### DISCUSSION TITLE
EA Launcher Update Loop

### DISCUSSION THREAD

**Discussion by rickneworleansla** [moraroy/NonSteamLaunchers-On-Steam-Deck#170] [Q&A] [answered]
What is the best way to update the EA launcher after using the script?  I installed EA and a few other nonsteam launchers a few weeks ago.  I recently started to get prompted to update the EA launcher.  It downloads the update then forces you to restart the app.  After the app restarts it asks for the update again and forces another restart.  This is basically a loop that never lets you get to use the launcher again.  

I thought of removing and reinstalling everything or just the EA launcher using the script which brought up more questions.  If I use the script to remove the EA launcher would that effect the EA games I've already installed using the EA launcher?  Should I download the latest version of NonSteamLauncher and use that newest one to remove and re-add the EA launcher?  Do newer versions of the NonSteamLauncher script work with currently installed launchers for removal and re-adding?  I may want to do this in the future anyway to get new NonSteamLauncher features or bug fixes.

I found a fix for the boot loop that suggest downloading the new EA launcher and changing the target in Steam for it to update then back to the original target after it updates.  The paths are different though when using NonSteamLauncher and there is the extra EA launch option when using NonSteamLauncher.  I'm not sure how all that should be set when trying this method.

Hopefully it can be as simple as updating the NonSteamLauncher and removing and re-adding EA.? ..but not mess up the installed games.

Sorry if this was asked already.  I couldn't find anything when searching the last few days.  Thanks in advance!!

**Comment by moraroy** [comment, marked as answer]
@rickneworleansla as a rule of thumb for me personally, if a Launcher is giving me problems I try to let it run and update itself in Desktop Mode first. I let it restart itself and if it doesn't come back I launch it again through steam. NSL doesn't update any of the launchers for you in any way. The best thing to do is always make sure you're using the latest Proton GE which the script does use, and if you really need to, you can "Re-install" the launchers by check marking it again, and installing. This shouldnt erase your games. When it comes to the EA App I actually dont have it installed on my deck I still use origin... Origin doesmt work anymore though....so my installation is still working but origin can no longer be installed at all. The EA App is tricky sometimes. So as of right now the best thing to do is troubleshoot until you can get it stable and updated.

**Comment by rickneworleansla** [comment]
Thanks for the quick response and solution!  I had previously tried desktop mode and Proton experimental but that still resulted in the update loop.  I took you're suggestion of simply reinstalling the EA launcher using the NSL checkbox.  I then ran EA once in desktop mode and then after the update ran it worked again.   Everything seems to work fine once again and the games were unaffected.   I made sure to turn off auto update in the EA Launcher settings so it does not happen again.

I was still curious though, should I download and use the latest NSL version going forward?  Will the newer version work the same for removing or reinstalling the launchers I already have installed with the old NSL script?

**Reply by moraroy** [reply]
@rickneworleansla as of right now you only ever need the .Desktop file.. the desktop file immediately grabs the latest version, for the foreseeable future this shouldnt effect anything because the script automatically updates itself.
